lib/IRGen/GenCast.cpp

```diff
@@ -329,12 +329,13 @@ static llvm::Function *emitExistentialScalarCastFn(IRGenModule &IGM,
   }
   
   llvm::Type *returnTy = llvm::StructType::get(IGM.getLLVMContext(), returnTys);
-  
+
   auto fnTy = llvm::FunctionType::get(returnTy, argTys, /*vararg*/ false);
   auto fn = llvm::Function::Create(fnTy, llvm::GlobalValue::PrivateLinkage,
                                    llvm::Twine(name), IGM.getModule());
+  // FIXME: should we be setting the visibility and the DLL storage as well?
```

Comment:
Hidden visibility isn't meaningful for internal / private entities, and I believe LLVM considers it a consistency error to have that combination.  There's no need to make this explicit in the code.